### ISSUE TITLE
Improved Upload Component

### ISSUE THREAD

**Issue by lgalke** [locdb/locdb-frend#168]
The upload component received a major update. Most of the changes improved the code quality in general, but some also affect the behaviour.

Status is now color coded:

- Red : error
- Yellow: missing information
- Green: ready for upload

Plus the time for uploading is greatly reduced! 💯 

Please assert that everything still works as expected.

**Comment by lgalke** [comment]
f3373a87b4e769cc3b8d5d9009dd15595cd748e4 for reference

**Comment by LauraErhard** [comment]
The color works fine and the upload speed is noticeable faster.
But I got an error as I tried to upload the collection "Kommunalpolitik - was uns prägt - was uns eint" (PPN 491186231).
The error is as follows:
`Send Scans failed:  Object { identifier: Object, firstpage: "50", lastpage: "50", file: File, resourceType: "COLLECTION", status: null, uploading: false, textualPdf: false, err: "400 - Bad Request {}" } 400 - Bad Request {}  scan.component.ts:273:4
File.lastModifiedDate sollte nicht mehr verwendet werden. Verwenden Sie File.lastModified stattdessen.`

**Comment by LauraErhard** [comment]
Something curious happened now:
I uploaded the collection "Teaching visual methods in the social sciences" (PPN 490989020) the file upload component guessed correctly and set the resource type to "collection":
![screenshot-2017-11-10 loc-db frontend 1](https://user-images.githubusercontent.com/32642653/32647554-3020f6b2-c5f2-11e7-84f6-a6b469b270d0.png)

But in the "OCR unprocessed tab the type is now "monograph":
![screenshot-2017-11-10 loc-db frontend](https://user-images.githubusercontent.com/32642653/32647595-571704c8-c5f2-11e7-8250-9770f699d6fa.png)

**Comment by lgalke** [comment]
Regarding the first problem: `Bad Request` is usually issued when the PPN could not be found.

Regarding the second one, I will investigate whether the type is submitted correctly.

**Comment by LauraErhard** [comment]
first problem: I checked the SWB and the PPNs are there and correct, e.g.:
![screenshot-2017-11-10 swb online-katalog - results availability 1](https://user-images.githubusercontent.com/32642653/32653144-0fdaae88-c607-11e7-9eae-c31fea42d56d.png)

**Comment by lgalke** [comment]
Could partly reproduce the 2nd problem with `492932506.43-52.pdf`. In the first place, I experienced the exact same behaviour. 492932506 was shown as `MONOGRAPH` despite having sub-resources and a scan attached to them. Then I removed both the top-level resource and the lower level resource and added it again. In this case, the resource was correctly shown as `COLLECTION`.

If a resource is not explicitly deleted, it is kept in the database (and should be) but not shown in the ToDo list because there is no scan with unlinked references attached. After deleting the resource as a whole and uploading again, the resource type is set correctly. The assumption is, that someone (most likely me) manually set the type of the resource to `MONOGRAPH` and it never got reset.
